### PR TITLE
Henrykie/helix swarm

### LINKED ISSUE #41: Feature request: Helix Swarm automated deployment of ElastiCache serverless
### Use case

Helix Swarm currently provisions a Redis container alongside the Helix Swarm container. We should provide a flag to enable ElastiCache serverless instead.

### Solution/User Experience

`enable_elasticache_serverless` variable on the `helix-swarm` module.

### Alternative solutions

_No response_

### LINKED ISSUE #294: Bug: Helix Swarm module should support SSO logins
### Expected Behaviour

There should be an option in `helix-swarm` module to enable SSO when creating the service.

### Current Behaviour

There is no option currently. 

### Possible Solution

To enable, you have to add the following line to the `config.php` file:

```php
        'p4' => array(
            // ...
            'sso'       => 'enabled', 
         ),
```

### Steps to Reproduce

Install the `helix-swarm` module and load it up in the browser. 

### Cloud Game Development Toolkit version

latest

### PR DESCRIPTION
**Pull request by henrykie**
**Issue number:**
closes #41 
closes #294 

## Summary

General improvements to the Perforce Helix Swarm module, and Perforce Helix Swarm example configurations.

### Changes

> Please provide a summary of what's being changed

= Replace Redis sidecar container with Elasticache.
= Adds SSO enable flag to Helix Swarm module.
- Minor changes to formatting on HAS module.
- Adds example configuration that provisions Helix Core, Helix Swarm, and Helix Authentication Service.

### User experience

> Please share what the user experience looks like before and after this change

- Users should specify `enable_sso` to configure Swarm to use the SSO login page.
- Redis is no longer configured as a sidecar in the Helix Swarm module. Instead, Helix Swarm now uses an Elasticache cluster it provisions itself. If you would like to use an external redis connection this can be passed through the `existing_redis_connection` variable.
- Users can refer to the "complete" example under the Perforce module for a Helix Core, Helix Swarm, Helix Authentication Service configured deployment.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented

<details>
<summary>Is this a breaking change?</summary>

Yes. This does replace the existing Redis sidecar container in the Helix Swarm module with a dedicated Elasticache cluster.

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created might not be successful.

### CODE REVIEW

File: modules/perforce/examples/complete/main.tf [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+module "perforce_helix_authentication_service" {
+  source                                   = "../../helix-authentication-service"
+  vpc_id                                   = aws_vpc.perforce_vpc.id
+  cluster_name                             = aws_ecs_cluster.perforce_cluster.name
+  helix_authentication_service_alb_subnets = aws_subnet.public_subnets[*].id
+  helix_authentication_service_subnets     = aws_subnet.private_subnets[*].id
+  certificate_arn                          = aws_acm_certificate.helix.arn
+
+  enable_web_based_administration = true
+  fqdn                            = "https://auth.helix.${var.root_domain_name}"
```
Comment (gabebatista) [inline modules/perforce/examples/complete/main.tf:61]: You used `core.helix.perforce.${var.root_domain_name}` for helix core, so should this one be `auth.helix.perforce.${var.root_domain_name}` to keep the naming standard?

**Review by gabebatista — APPROVED** [review]
lgtm, made some minor comments on naming standard for fqdn prefix

File: docs/media/images/helix-swarm-architecture.jpg
Comment (kylesomers) [inline docs/media/images/helix-swarm-architecture.jpg:1]: Cut dimensions in half. Renders really large in the docs

File: modules/perforce/helix-authentication-service/main.tf [hunk 2]
```diff
@@ -84,10 +84,9 @@ resource "aws_ecs_task_definition" "helix_authentication_service_task_definition
           value = var.enable_web_based_administration ? "true" : "false"
         },
         {
-          name = "TRUST_PROXY"
+          name  = "TRUST_PROXY"
```
Comment (kylesomers) [inline modules/perforce/helix-authentication-service/main.tf:87]: What's going on here with the spacing?

**Review by kylesomers — CHANGES_REQUESTED** [review]
see inline comments. small fixes

File: modules/perforce/helix-authentication-service/main.tf (hunk 2 above)
Comment (henrykie) [reply modules/perforce/helix-authentication-service/main.tf:87]: Formatting. My machine aligns spacing for fields in the same block. Other folks do not. We need to add a prettier config.

File: docs/media/images/helix-swarm-architecture.jpg
Comment (henrykie) [reply docs/media/images/helix-swarm-architecture.jpg:1]: Understood.

[the remark this answers is not in the document]
Comment (henrykie) [reply docs/media/images/helix-swarm-architecture.jpg:1]: Will fix.

**Comment by henrykie** [comment]
@jorisdon @gabebatista @kylesomers I've made changes to the actual commit history on this branch to match Joris's git strategy as discussed. Individual commits can be reviewed as discrete pieces (swarm updates, core updates, etc...). This makes reviews easier. It also allows us to rebase the feature branch onto main and have each of these commits show up in changelog.

File: modules/perforce/examples/complete/main.tf (hunk 1 above)
Comment (henrykie) [reply modules/perforce/examples/complete/main.tf:61]: Good catch. I will adjust.

File: docs/modules/perforce/examples/complete.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Perforce Complete Example
+
+This example provisions [Helix Core](https://www.perforce.com/products/helix-core), [Helix Swarm](https://www.perforce.com/products/helix-swarm), and the [Helix Authentication Service](https://www.perforce.com/downloads/helix-authentication-service). It also configures security groups for each of these modules to allow inter-service communication. This example takes a single input variabl:`root_domain_name` is expected to correspond to an existing [AWS Route53 hosted zone](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/route-53-concepts.html#route-53-concepts-hosted-zone). This hosted zone is used for provisioning DNS records used for external and internal routing, and enables this example to create validated SSL certificates on your behalf.
```
Comment (jorisdon) [inline docs/modules/perforce/examples/complete.md:3]: typo fix: variabl -> variable

**Review by jorisdon — DISMISSED** [review]
LGTM

**Review by kylesomers — APPROVED** [review]

**Pull request merged**